frontend/src/metabase/api/entity-id.ts

```diff
@@ -0,0 +1,85 @@
+import { type BaseEntityId, isNanoID } from "metabase-types/api/entity-id";
+
+import { Api } from "./api";
+
+const validEntityTypes = [
+  "action",
+  "card",
+  "collection",
+  "dashboard",
+  "dashboard-card",
+  "dashboard-tab",
+  "dataset",
+  "dimension",
+  "metric",
+  "permissions-group",
+  "pulse",
+  "pulse-card",
+  "pulse-channel",
+  "segment",
+  "snippet",
+  "timeline",
+  "user",
+] as const;
+
+export type EntityType = typeof validEntityTypes[number];
+
+type TranslateEntityIdRequest = Partial<Record<EntityType, BaseEntityId[]>>;
+
+export type TranslateEntityIdResponseSuccess = {
+  status: "success";
+  id: string | number;
+  type: EntityType;
+};
+
+export type TranslateEntityIdResponseNotFound = {
+  status: "not-found";
+  type: EntityType;
+  id: null;
+};
+
+export type TranslateEntityIdResponse = Record<
+  BaseEntityId,
+  TranslateEntityIdResponseSuccess | TranslateEntityIdResponseNotFound
+>;
+
+type TranslateEntityIdError = {
+  message: string;
+  explanation: Record<EntityType, [[string]]>;
+};
+
+const entityIdApi = Api.injectEndpoints({
+  endpoints: builder => ({
+    translateEntityId: builder.query<
+      TranslateEntityIdResponse,
+      TranslateEntityIdRequest
+    >({
+      query: (requestEntities: TranslateEntityIdRequest) => {
+        Object.values(requestEntities)
+          .flat()
+          .forEach(entityId => {
+            if (isNanoID(entityId)) {
+              throw new Error(`Entity ID ${entityId} must be a NanoID`);
+            }
```

Comment:
Should this be `if (!isNanoID(entityId))` instead?

Reply:
woah, you reviewed too quickly 😂

Reply:
whoops, my bad 😆

Reply:
issue is resolved 😄